src/main/java/de/tum/in/www1/artemis/service/ZipFileService.java

```diff
@@ -0,0 +1,81 @@
+
+    /**
+     * Create a zip file of the given paths and save it in the zipFilePath
+     *
+     * @param zipFilePath path where the zip file should be saved
+     * @param paths multiple paths that should be zipped
+     * @throws IOException if an error occurred while zipping
+     */
+    public void createZipFile(Path zipFilePath, List<Path> paths) throws IOException {
+        try (ZipOutputStream zipOutputStream = new ZipOutputStream(Files.newOutputStream(zipFilePath))) {
+            paths.stream().filter(path -> !Files.isDirectory(path)).forEach(path -> {
+                ZipEntry zipEntry = new ZipEntry(path.toString());
+                copyToZipFile(zipOutputStream, path, zipEntry);
+            });
+        }
+    }
+
+    /**
+     * Create a zip file of the given paths and save it in the zipFilePath
+     *
+     * @param zipFilePath path where the zip file should be saved
+     * @param paths multiple paths that should be zipped
+     * @param pathsRoot the root path relative to <code>paths</code>
+     * @throws IOException if an error occurred while zipping
+     */
+    public void createZipFile(Path zipFilePath, List<Path> paths, Path pathsRoot) throws IOException {
+        try (ZipOutputStream zipOutputStream = new ZipOutputStream(Files.newOutputStream(zipFilePath))) {
+            paths.stream().filter(path -> !Files.isDirectory(path)).forEach(path -> {
+                ZipEntry zipEntry = new ZipEntry(pathsRoot.relativize(path).toString());
+                copyToZipFile(zipOutputStream, path, zipEntry);
+            });
+        }
+    }
+
+    /**
+     * Recursively include all files in contentRootPath and create a zip file 'zipFileName' in the folder 'zipFileFolderName'
+     *
+     * @param zipFilePath path where the zip file should be saved
+     * @param contentRootPath a path to a folder: all content in this folder (and in any subfolders) will be included in the zip file
+     * @return the path of the newly created zip file for further processing
+     * @throws IOException if an error occurred while zipping
+     */
+    public Path createZipFileWithFolderContent(Path zipFilePath, Path contentRootPath) throws IOException {
+        try (ZipOutputStream zipOutputStream = new ZipOutputStream(Files.newOutputStream(zipFilePath))) {
+            Files.walk(contentRootPath).filter(path -> !Files.isDirectory(path)).forEach(path -> {
+                ZipEntry zipEntry = new ZipEntry(contentRootPath.relativize(path).toString());
+                copyToZipFile(zipOutputStream, path, zipEntry);
+            });
+        }
+        return zipFilePath;
+    }
+
+    private void copyToZipFile(ZipOutputStream zipOutputStream, Path path, ZipEntry zipEntry) {
+        try {
+            zipOutputStream.putNextEntry(zipEntry);
+            Files.copy(path, zipOutputStream);
+            zipOutputStream.closeEntry();
+        }
+        catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968852)